src/test/java/ch/jalu/configme/resource/YamlFileResourceRootMapPropertyTest.java

```diff
@@ -0,0 +1,78 @@
+package ch.jalu.configme.resource;
+
+import ch.jalu.configme.SettingsHolder;
+import ch.jalu.configme.TestUtils;
+import ch.jalu.configme.configurationdata.ConfigurationData;
+import ch.jalu.configme.configurationdata.ConfigurationDataBuilder;
+import ch.jalu.configme.properties.MapProperty;
+import ch.jalu.configme.properties.Property;
+import ch.jalu.configme.properties.types.BeanPropertyType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static ch.jalu.configme.properties.PropertyInitializer.newListProperty;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+class YamlFileResourceRootMapPropertyTest {
+
+    @TempDir
+    public Path temporaryFolder;
+
+    @Test
+    void shouldWriteAndReadFile() {
+        Path yamlFile = TestUtils.createTemporaryFile(temporaryFolder);
+
+        YamlFileResource resource = new YamlFileResource(yamlFile);
+        ConfigurationData configurationData = ConfigurationDataBuilder.createConfiguration(SampleConfig.class);
+
+        configurationData.setValue(SampleConfig.SHAPE, singletonList("foo"));
+
+        Map<String, InnerProperties> mapProperty = new HashMap<>();
+        mapProperty.put("props", new InnerProperties());
+        mapProperty.get("props").setProps(Arrays.asList("bar", "baz"));
+
+        configurationData.setValue(SampleConfig.MAP_PROPERTY, mapProperty);
+
+        resource.exportProperties(configurationData);
```

Comment:
Good rewrite to only use the YamlFileResource! 👏 

To make sure what we exported can be read again, we should call Property#determineValue instead of using the ConfigurationData. So for the line below (L48), I think it should become the following:

```java
PropertyValue<Map<String, InnerProperties>> actualPropertyValue =
    SampleConfig.MAP_PROPERTY.determineValue(resource.createReader());
```

The `PropertyValue` gives us two things: whether the representation of the property was "fully valid" in the resource, which we expect to be true, and the value, which we expect to be the same as `mapProperty`: 
```java
        assertThat(actualPropertyValue.isValidInResource(), equalTo(true));
        assertThat(actualPropertyValue.getValue(), equalTo(mapProperty));
```

Changing the test like this will make it fail, because `mapProperty` has a different instance of `InnerProperties` than the given one, and `InnerProperties` doesn't define an `equals` method. You can either define a proper `equals()` method on InnerProperties to make the assertion work, or check the map's values manually:
```java
        assertThat(actualPropertyValue.isValidInResource(), equalTo(true));
        assertThat(actualPropertyValue.getValue().keySet(), contains("props"));
        assertThat(actualPropertyValue.getValue().get("props").getProps(), contains("bar", "baz"));
```

I personally don't have any preference—if I wrote this test, I would check the key set and values manually as in the last snippet but just do whatever you prefer.

PS: please add `given` / `when` / `then` comments as seen in other test methods (`given` should be before L32, `when` before L47 because that's what is being tested, and `then` before L50). Thanks!